app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,70 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all.order(:completion_date).reverse_order
+  end
+
+  def show
+    id = params[:id].to_i
+    @task = Task.find_by(id: id)
+
+    if @task.nil?
+      render :notfound, status: :not_found
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def edit
+    @task = Task.find(params[:id].to_i)
```

Comment:
What happens if the task isn't found? You should include error-handling code in all of these actions that look for a single task.

Reply:
I fixed this in the code. I see that I have similar actions that look for a single task in update, complete, and destroy - there should be error handling in each of these too? It seems like complete and destroy and update already rely on id (as in there is an ID), so do I still need error handling for these cases?